src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,140 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import React, {
+  useContext, useState,
+} from 'react';
+import cn from 'classnames';
+import { Todo } from '../../types/Todo';
+import { TodosContext, TodosUpdateContext } from '../../contexts/TodosProvider';
+import { TodoAction } from '../../types/TodoAction';
+import { customDebounce } from '../../utils/useDebounce';
+import { EditTodoForm } from '../EditTodoForm';
+
+interface Props {
+  todo: Todo,
+  isLoading?: boolean
+}
+
+export const TodoItem: React.FC<Props> = React.memo(
+  ({ todo, isLoading = false }) => {
+    const [isEditing, setIsEditing] = useState(false);
+    const { deleteTodo } = useContext(TodosUpdateContext);
+    const { dispatch, operatingTodos } = useContext(TodosContext);
+    const { title, completed, id } = todo;
+
+    // const handleTodoChanged = (newTitle: string) => {
+    //   const normalizedTitle = newTitle.trim();
+
+    //   if (normalizedTitle !== title) {
+    //     if (normalizedTitle === '') {
+    //       dispatch({
+    //         type: TodoAction.Delete,
+    //         todo,
+    //       });
+    //     } else {
+    //       dispatch({
+    //         type: TodoAction.Update,
+    //         todo: {
+    //           ...todo,
+    //           title: normalizedTitle,
+    //         },
+    //       });
+    //     }
+    //   }
+
+    //   setIsEditing(false);
+    // };
+
+    const handleTodoChangeCancelled = () => {
+      setIsEditing(false);
+    };
+
+    // const handleTodoStatusChanged = () => {
+    //   dispatch({
+    //     type: TodoAction.Update,
+    //     todo: {
+    //       ...todo,
+    //       completed: !todo.completed,
+    //     },
+    //   });
+    // };
+
+    const handleDblclick = customDebounce(() => {
+      if (!isEditing) {
+        setIsEditing(true);
+      }
+    }, 300);
```

Comment:
You dont need to use deboune, in mouse event there is in-build parametr about click count. But anyway - cool idea!